Tests/MeiliSearchUnitTests/IndexesTests.swift

```diff
@@ -277,6 +277,45 @@ class IndexesTests: XCTestCase {
     self.wait(for: [expectation], timeout: 10.0)
   }
 
+  func testDeleteIndexIfExists() {
+    // Prepare the mock server
+    session.pushEmpty(code: 204)
+
+    // Start the test with the mocked server
+    let expectation = XCTestExpectation(description: "Delete Movies index")
+
+    self.index.deleteIfExists { result in
+      if result {
+        XCTAssertTrue(result)
+        expectation.fulfill()
+      } else {
+        XCTFail("Failed to delete Movies index, it was not present on the server")
+      }
+      expectation.fulfill()
+    }
+
+    self.wait(for: [expectation], timeout: 10.0)
+  }
+
+  func testDeleteIndexIfExistsWhenIsnt() {
+    // Prepare the mock server
+    session.pushEmpty(code: 404)
+
+    // Start the test with the mocked server
+    let expectation = XCTestExpectation(description: "Delete Movies index only if exists")
+
+    self.index.deleteIfExists { result in
+      if !result {
+        XCTAssertFalse(result)
+        expectation.fulfill()
+      } else {
+        XCTFail("Failed to delete Movies index, it was not present on the server")
```

Comment:
```suggestion
        XCTFail("Deleting the index should have returned false as the index does not exist on the server")
```

Reply:
@bidoubiwa thanks for your comment, I tried to find something like this https://betterspecs.org/ for swift but unfortunately I only found some random articles about this. Do you have something like that to share?

Reply:
@brunoocasali  this is a very interesting ressource. I'm not the best at writing good and consistent specs. I will definitely give it a look as it would mean to change very spec description of the repository

Reply:
Oh, actually it was not a criticism ok? It's because in ruby community we have "guides" everywhere haha an I like to follow them to evict "bikeshedding", but and I really liked what you did on this project :)

Reply:
Oh no don't worry, I totally agree that we should follow some specs best practices! I didn't take it as a criticism but as a good resource that I want to follow :)